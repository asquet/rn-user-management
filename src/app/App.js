import React from 'react';
import { Provider } from 'react-redux';

import UiRoot from './UiRoot';
import store from './redux/store';

export default function App() {
  return (
    <Provider store={store} >
      <UiRoot />
    </Provider>
  );
}
