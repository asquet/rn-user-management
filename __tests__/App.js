import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/app/App';

// Note: test renderer must be required after react-native.

describe('Application', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
