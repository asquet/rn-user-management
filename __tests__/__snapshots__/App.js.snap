// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          elevation={0}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "left": -603,
                "position": "absolute",
                "shadowColor": "#000",
                "shadowOpacity": 0.8,
                "shadowRadius": 3,
                "top": 0,
              },
              Object {
                "height": 1334,
                "width": 600,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "marginBottom": 20,
                }
              }
            >
              Menu
            </Text>
            <View
              style={
                Object {
                  "flex": 0,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Users
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Positions
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Roles
                </Text>
              </View>
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          />
        </View>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "left": -3,
                "paddingLeft": 3,
                "position": "absolute",
                "top": 0,
              },
              Object {
                "height": 1334,
                "width": 753,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                },
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#707070",
                    "fontSize": 18,
                  },
                  Object {
                    "fontSize": 30,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Hello, dear friend!
            </Text>
            <View
              style={
                Object {
                  "flex": 0,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Users
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Positions
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 5,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Roles
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginLeft": 15,
                      "marginRight": 15,
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "shadowColor": "rgba(0,0,0, .4)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 1,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 1,
                    },
                    Object {
                      "padding": 12,
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 3,
                        "padding": 5,
                      },
                      Object {
                        "backgroundColor": "#0aaeff",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": undefined,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  MoveOn
                </Text>
              </View>
            </View>
          </View>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
